src/actions/actions.ts

```diff
@@ -327,15 +327,19 @@ export class ActionPayloadTypes {
      * @param onSuccessAction Any other action
      * @param confirm params for confirm modal
      * @param bcKey key called bk
+     * @param customParam custom any parameter
      */
     sendOperation: {
         bcName: string,
         operationType: OperationTypeCrud | string,
         widgetName: string,
         onSuccessAction?: AnyAction,
         confirm?: string,
-        // TODO: Will not be optional in 2.0.0
+        /**
+         * @deprecated TODO: Remove in 2.0.0 in favor of customParam
+         */
         bcKey?: string,
+        [customParam: string]:any,
```

Comment:
This will not be merged.

1) When introducing `[someVar: someType]: any` pattern to your type it is important to wrap into into a separate property:
```tsx
interface someInterface {
  // ...
  customProperties: Record<string, any>
}
```
Otherwise you will be open for misspelling mistakes:
```tsx
sendOperation({ onSucessAction })
// no error will be thrown because `onSucessAction` now considered `[customParam: string]`  
```
2) This is an overall incorrect approach to allow core actions have unspecified payload and bind customization logic to the fact of the presence of said payload.  
When client app wants an additional payload parameter it is the client app responsibility to extend core type and dispatch an extended actions instead of the core one:

```tsx
import {ActionPayloadTypes} from '@tesler-ui/core'
export class CustomActionPayloadTypes extends ActionPayloadTypes {
  // ...
  sendOperation: ActionPayloadTypes ['sendOperation'] & {
        customParam: number
    } = z
}
```
And then build observable action creators from this class and then dispatch extended action like this:
```tsx
dispatch($clientDo.sendOperation({
  // core payload
  // ...
  customParam: number
})
```
And then handle this additional parameter in a client side reducer.  
No additional code in Tesler UI is necessary for that.

Reply:
custom Param moved to client application